### ISSUE TITLE
Refactor lib/monitor

### ISSUE THREAD

**Issue by Liamsi** [dedis/cothority#163]
1) get rid of global vars, for instance: in [measure.go](https://github.com/dedis/cothority/blob/b682ebe6749cb37844b44575db4342b985e054b7/lib/monitor/measure.go#L23-L32) and make sure that deployment and everything else still works (use functions & structs instead of global vars to communicate data across modules)

2) Find and understand pitfalls like this: What is the difference between: [`StopSink`](https://github.com/dedis/cothority/blob/b682ebe6749cb37844b44575db4342b985e054b7/lib/monitor/measure.go#L63) and [`End`](https://github.com/dedis/cothority/blob/b682ebe6749cb37844b44575db4342b985e054b7/lib/monitor/measure.go#L172) (both try to close a connection which is stored in a global var). 

In other words: Refactor until you have a clear API and change tests to show how-to use this API.

**Comment by ineiti** [comment]
#284
